### PR TITLE
[INTEL MKL] Fix memory leak issue in MklFusedBatchNorm

### PR DESCRIPTION
**Pull request by jzhoulon**
fix the memory leak bug in mkl batch norm, free the temp weight buffer used for MKL-DNN weight memory primitive.

### CODE REVIEW

**Review by tatianashp — APPROVED** [review]
Looks good.  Thanks for fixing.

**Comment by agramesh1** [comment]
@harshini-gadige can you take a look at this PR? the one failure //tensorflow/c/eager:c_api_test_gpu seems to be unrelated and is failing in other PRs also.

**Comment by harshini-gadige** [comment]
> @harshini-gadige can you take a look at this PR? the one failure //tensorflow/c/eager:c_api_test_gpu seems to be unrelated and is failing in other PRs also.

Sure. Will look into this.

**Comment by agramesh1** [comment]
@harshini-gadige The failures are not related to this PR,  other PRs are also failing the same tests. Can you take a look at it again? Thanks.

**Comment by harshini-gadige** [comment]
> @harshini-gadige The failures are not related to this PR, other PRs are also failing the same tests. Can you take a look at it again? Thanks.

Looking into this. It takes time and I'll keep posted. Thanks !

**Comment by harshini-gadige** [comment]
@tatianashp  Can I bypass these 2 windows bazel failures and proceed with "Ready to pull"  ?

**Comment by harshini-gadige** [comment]
> @tatianashp Can I bypass these 2 windows bazel failures and proceed with "Ready to pull" ?

As per the response I got from the chat, proceeding to get this merged.
